mindsdb/integrations/handlers/rag_handler/settings.py

```diff
@@ -0,0 +1,460 @@
+)
+from langchain import Writer
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+from langchain.docstore.document import Document
+from langchain.document_loaders import DataFrameLoader
+from langchain.embeddings.base import Embeddings
+from langchain.embeddings.huggingface import HuggingFaceEmbeddings
+from langchain.vectorstores import FAISS, Chroma, VectorStore
+from pydantic import BaseModel, Extra, Field, validator
+from writer.models import shared
+
+DEFAULT_EMBEDDINGS_MODEL = "BAAI/bge-base-en"
+
+SUPPORTED_VECTOR_STORES = ("chroma", "faiss")
+
+SUPPORTED_LLMS = ("writer", "openai")
+
+## Default parameters for RAG Handler
+
+# this is the default prompt template for qa
+DEFAULT_QA_PROMPT_TEMPLATE = """
+Use the following pieces of context to answer the question at the end. If you do not know the answer,
+just say that you do not know, do not try to make up an answer.
+Context: {context}
+Question: {question}
+Helpful Answer:"""
+
+# this is the default prompt template for if the user wants to summarize the context before qa prompt
+DEFAULT_SUMMARIZATION_PROMPT_TEMPLATE = """
+Summarize the following texts for me:
+{context}
+
+When summarizing, please keep the following in mind the following question:
+{question}
+"""
+
+DEFAULT_CHUNK_SIZE = 500
+DEFAULT_CHUNK_OVERLAP = 50
+DEFAULT_VECTOR_STORE_NAME = "chroma"
+DEFAULT_VECTOR_STORE_COLLECTION_NAME = "collection"
+DEFAULT_PERSISTED_VECTOR_STORE_FOLDER_NAME = "vector_store_folder"
+
+
+def is_valid_store(name) -> bool:
+    return name in SUPPORTED_VECTOR_STORES
+
+
+class VectorStoreFactory:
+    """Factory class for vector stores"""
+
+    @staticmethod
+    def get_vectorstore_class(name):
+
+        if not isinstance(name, str):
+            raise TypeError("name must be a string")
+
+        if not is_valid_store(name):
+            raise ValueError(f"Invalid vector store {name}")
+
+        if name == "faiss":
+            return FAISS
+
+        if name == "chroma":
+            return Chroma
+
+
+def get_chroma_settings(persist_directory: str = "chromadb") -> Settings:
+    """Get chroma settings"""
+    return Settings(
+        chroma_db_impl="duckdb+parquet",
+        persist_directory=persist_directory,
+        anonymized_telemetry=False,
+    )
+
+
+def get_available_writer_model_ids(args: dict) -> list:
+    """Get available writer LLM model ids"""
+
+    writer_client = writer.Writer(
+        security=shared.Security(
+            api_key=args["writer_api_key"],
+        ),
+        organization_id=args["writer_org_id"],
+    )
+
+    res = writer_client.models.list(organization_id=args["writer_org_id"])
+
+    available_models_dict = json.loads(res.raw_response.text)
+
+    return [model["id"] for model in available_models_dict["models"]]
+
+
+def get_available_openai_model_ids(args: dict) -> list:
+    """Get available openai LLM model ids"""
+
+    openai.api_key = args["openai_api_key"]
+
+    res = openai.Engine.list()
```

Comment:
AFAIK this handler only should support user specific openai key. As in if no api key is specified it will fail and not attempt to load from env i.e. public key.